### PR TITLE
Extend version to four numbers (5.x.y.z)

### PR DESCRIPTION
**Pull request by kontura**
Unfortunately this is a breaking change because it is changing a struct in `include/libdnf5/version.hpp`

Therefore I have targeted the `5.2.0.0` branch.
This means it will not get into rawhide before dnf5 is made default.

I have called the new version number `PRIME` but I am open to any other naming.

### CODE REVIEW

File: dnf5/include/dnf5/version.hpp [hunk 1]
```diff
@@ -27,6 +27,7 @@ namespace dnf5 {
 /// Application version
 /// @since 5.0
 struct ApplicationVersion {
+    std::uint16_t prime;
```
Comment (ppisar) [inline dnf5/include/dnf5/version.hpp:30]: Could you document a meaning of the version fields here in the header file? People studying API reads the header files, not the commit messages.

**Comment by ppisar** [comment]
Your last commit broke packit:

~~~~
2024-03-01 13:26:25.665 base_git.py       INFO   Using user-defined script for ActionName.get_current_version: [['rpmspec', '-q', '--queryformat', '%{VERSION}\\n', '*spec', '|head', '-n1']]
2024-03-01 13:26:25.665 commands.py       DEBUG  Command: rpmspec -q --queryformat %{VERSION}\n *spec |head -n1
2024-03-01 13:26:25.668 logging.py        INFO   rpmspec: -n1: unknown option
2024-03-01 13:26:25.669 commands.py       ERROR  Command 'rpmspec -q --queryformat %{VERSION}\\n *spec |head -n1' failed.
2024-03-01 13:26:25.669 commands.py       DEBUG  Command stderr: rpmspec: -n1: unknown option
~~~~

**Comment by kontura** [comment]
In an attempt to avoid future conflicts I moved the lines around a bit, it should match: https://github.com/rpm-software-management/dnf5/pull/1296

Anyway the PR should be ready and the builds should work now.

**Comment by ppisar** [comment]
The failing test "dnf/installonly.feature:414  Handle over-limit custom kernel without installonlypkg(kernel) provide" is unrelated. Otherwise, the package builds and the tests are fine.

**Review by ppisar — APPROVED** [review]

**Pull request merged**